addons/ui/org.openhab.ui.cometvisu/src/main/java/org/openhab/ui/cometvisu/internal/util/ClientInstaller.java

```diff
@@ -0,0 +1,306 @@
+                    }
+                }
+            }
+        } else {
+            logger.error("webfolder {} is no directory", webFolder.getAbsolutePath());
+        }
+    }
+
+    /**
+     * Search for a file in the known file structure of the CometVisu client.
+     * The search oder is:
+     *
+     * . # for releases
+     * build/ # for CometVisu >= 0.11 with generated build
+     * source/ # for CometVisu >= 0.11 without generated build (source version)
+     * src/ # for CometVisu < 0.11 (source version)
+     *
+     * @param webFolder {File} folder to start the search
+     * @param fileName {String} Filename to lookup
+     * @return
+     */
+    public static File findClientRoot(File webFolder, String fileName) {
+        File file = new File(webFolder, fileName);
+        if (!file.exists()) {
+            File build = new File(webFolder, "build");
+            File source = new File(webFolder, "source");
+            File src = new File(webFolder, "src");
+            if (build.exists()) {
+                // new CometVisu client >= 0.11.x
+                file = new File(build, fileName);
+            } else if (source.exists()) {
+                // new CometVisu client >= 0.11.x
+                file = new File(source, fileName);
+            } else if (src.exists()) {
+                file = new File(src, fileName);
+            }
+        }
+        return file;
+    }
+
+    /**
+     * Checks if the CometVisu client is missing, but the automatic download is blocked by configuration.
+     *
+     * @return {boolean}
+     */
+    public boolean isDownloadAvailableButBlocked() {
+        return downloadAvailableButBlocked;
+    }
+
+    /**
+     * Compare two SemVer version strings and return true if releaseVersion is newer then currentVersion
+     *
+     * @param releaseVersion {String} e.g 0.11.0
+     * @param currentVersion {String} e.g. 0.10.0
+     * @return {Boolean}
+     */
+    private boolean isNewer(String releaseVersion, String currentVersion) throws NumberFormatException {
+        logger.debug("checking if {} is newer than {}", releaseVersion, currentVersion);
+        Matcher release = semverPattern.matcher(releaseVersion);
+        Matcher current = semverPattern.matcher(currentVersion);
+        if (!release.matches()) {
+            throw new NumberFormatException("release version format error " + releaseVersion);
+        }
+        if (!current.matches()) {
+            throw new NumberFormatException("current version format error " + currentVersion);
+        }
+
+        for (int i = 1; i <= 3; i++) {
+            if (Integer.parseInt(release.group(i)) > Integer.parseInt(current.group(i))) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Fetch the latest release description for the CometVisu project from github
+     *
+     * @return {Map}
+     */
+    private Map getLatestRelease() {
```

Comment:
The map is the parsed JSON response from the github API: https://developer.github.com/v3/repos/releases/#get-the-latest-release